src/graphtypes/simplegraphs/simpledigraph.jl

```diff
@@ -116,6 +116,61 @@ function add_edge!(g::SimpleDiGraph, e::SimpleDiGraphEdge)
 end
 
 
+function add_edges!(g::SimpleDiGraph, es)
+    T = eltype(g)
+    nvnew = nvold = nv(g)
+    neold = ne(g)
+    ftouched = falses(nvold)
+    btouched = falses(nvold)
+
+    for e in es
+        s, d = T.(Tuple(e))
+        if s < 1 || d < 1
+            error("Nonpositive vertex number")
+        end
+        for v in (s, d)
+            if v > nvnew
+                resize!(g.fadjlist, v)
+                resize!(g.badjlist, v)     
+                for i in nvnew+1:v
+                    g.fadjlist[i] = Vector{T}()
+                    g.badjlist[i] = Vector{T}()
+                end
+                nvnew = v
+            end
+        end
+        if s <= nvold
+            ftouched[s] = true
+        end
+        if d <= nvold 
+            btouched[d] = true   
+        end
+        append!(g.fadjlist[s], d)
+        append!(g.badjlist[d], s)
+    end
+    nenew = 0
+    for s in 1:nvold
+        if ftouched[s]
+            sort!(g.fadjlist[s])
+            uniquesorted!(g.fadjlist[s])
+        end
+        nenew += length(g.fadjlist[s])
+        if btouched[s]    
+            sort!(g.badjlist[s])
+            uniquesorted!(g.badjlist[s])
+        end
+    end
+    for s in nvold+1:nvnew
```

Comment:
Why not just mark the vertices in nvold+1:nvnew as touched and then avoid this loop?

Reply:
Because if many vertices are added, an unnecessary big vector is carried around.